modules/aks_node_pool/main.tf

```diff
@@ -5,7 +5,7 @@ resource "azurerm_kubernetes_cluster_node_pool" "autoscale_node_pool" {
   name                         = var.node_pool_name
   kubernetes_cluster_id        = var.aks_cluster_id
   vnet_subnet_id               = var.vnet_subnet_id
-  availability_zones           = var.availability_zones
+  zones                        = var.zones
```

Comment:
I know they've changed the `zone` type naming. But wanted to be sure there were other breaking changes in this update outside of us changing our variables that the user may have had as inputs into the scripts. Is this the case?

Reply:
In this upgrade there were a couple variables/blocks that were deprecated, few were renamed and few had change of type
| Original  | New  | Update  | Affects user input |
|---|---|---|---|
| availability_zones | zones | Name change only | Yes |
| nfs_raid_disk_zones |  nfs_raid_disk_zone | Instead of collection accepts only a single string value  | Yes |
| enforce_private_link_endpoint_network_policies | private_endpoint_network_policies_enabled | Name change only | Yes |
| enforce_private_link_service_network_policies  | private_link_service_network_policies_enabled | Name change only | Yes |
| georeplications | georeplications| property needs to be defined as block | No |
| role_based_access_control | role_based_access_control_enabled | block deprecated in favour of the new property | No|
| addon_profile | NA | block removed in favour of other properties | No |
| addon_profile.oms_agent | oms_agent | property updated as individual block | No |
| addon_profile.http_application_routing_enabled | http_application_routing_enabled | individual property | No |
| addon_profile.kube_dashboard | NA | Property removed | No |

Reply:
Ok, so several updates to existing user input correct? This is definitely a breaking change.